mangoPURE/data/transforms.py

```diff
@@ -0,0 +1,280 @@
+        self.noises_dataset = noise_dataset
+        self.snrs = snrs
+        self.no_noise_coef = no_noise_coef
+
+    def __call__(
+            self,
+            example: MixedExample,
+    ) -> MixedExample:
+        """
+        It fills the audio with random noise from dataset
+        """
+        random_pos = random.randint(0, len(self.noises_dataset) - 1)
+        noise_example = self.noises_dataset[random_pos]
+        noise_len = example.audio.shape[0]
+        noise_audio = expand_tensor(noise_example["audio"]["array"], noise_len)
+
+        noise_coef = calc_audio_adjustment_coef(example.audio, noise_audio, self.snrs, self.no_noise_coef)
+        example.audio += noise_audio * noise_coef
+
+        if example.noises_info is None:
+            example.noises_info = []
+        example.noises_info.append(SegmentInfo(
+            start=0,
+            length=noise_len,
+            class_id=noise_example["noise_id"]
+        ))
+        return example
+
+
+class AddRandomNoiseSegment(MixerTransform):
+    def __init__(
+            self,
+            noises_dataset: HfDataset,
+            snrs: tuple[float] = (10.0, 15.0, 20.0),
+            no_noise_coef: bool = True
+    ):
+        """
+        :param noises_dataset: the dataset of noises audio
+            It has to havc "audio":"array" and "noise_id" columns
+        :param snrs: __call__ generates random noise coefficient from this list
+        :param no_noise_coef: if False does not use noise coefficient at all
+        """
+        self.noises_dataset = noises_dataset
+        self.snrs = snrs
+        self.no_noise_coef = no_noise_coef
+
+    def __call__(
+            self,
+            example: MixedExample,
+    ) -> MixedExample:
+        """
+        Adds to the audio random noise from dataset at a random place
+        """
+        random_pos = random.randint(0, len(self.noises_dataset) - 1)
+        noise_example = self.noises_dataset[random_pos]
+        noise_audio = noise_example["audio"]["array"]
+        noise_len = noise_audio.shape[0]
+        audio_len = example.audio.shape[0]
+
+        if noise_len > audio_len:
+            noise_audio = noise_audio[:audio_len]
+            noise_len = audio_len
+
+        noise_coef = calc_audio_adjustment_coef(example.audio, noise_audio, self.snrs, self.no_noise_coef)
+
+        pos_start = random.randint(0, audio_len - 1)
+        if pos_start + noise_len <= audio_len:
+            example.audio[pos_start:pos_start + noise_len] += noise_coef * noise_audio
+            start, length = pos_start, noise_len
+        else:
+            end_len = audio_len - pos_start
+            start_len = noise_len - end_len
+            if start_len > end_len:
+                example.audio[:start_len] += noise_coef * noise_audio[-start_len:]
+                start, length = 0, start_len
+            else:
+                example.audio[-end_len:] += noise_coef * noise_audio[:end_len]
+                start, length = pos_start, end_len
```

Comment:
Code duplication, maybe we could add this one to utils ore somewhere else. Or add method like `choose_audio_position(audio, content)` that choses position for content with these rules.  And rewrite it  as
```
start1, start2, len = chose_audio_position(audio, noise)
example.audio[start1:start1+len]  += noise_coef * noise[start2:start2+len]
```

Or, if we use multiple tracks we can store even it's start end ends one track, and not add it